comit/src/expiries.rs

```diff
@@ -0,0 +1,631 @@
+        }
+
+        let (next_action, _state) = current_state.next();
+        next_action
+    }
+
+    /// Returns the recommended next action that Bob should take.
+    pub fn next_action_for_bob(&self, current_state: BobState) -> BobAction {
+        if current_state == BobState::Done {
+            return BobAction::NoFurtherAction;
+        }
+
+        let funded = current_state.has_funded_beta();
+
+        if self.beta_expiry_has_elapsed() {
+            if funded {
+                return BobAction::Refund;
+            }
+            return BobAction::Abort;
+        };
+
+        if !self.bob_has_time_to_complete(current_state) {
+            if funded {
+                return BobAction::WaitForRefund;
+            }
+            return BobAction::Abort;
+        }
+
+        let (next_action, _state) = current_state.next();
+        next_action
+    }
+
+    fn alpha_expiry_has_elapsed(&self) -> bool {
+        let now = self.alpha_connector.current_time();
+        now > self.alpha_expiry
+    }
+
+    fn beta_expiry_has_elapsed(&self) -> bool {
+        let now = self.beta_connector.current_time();
+        now > self.beta_expiry
+    }
+}
+
+/// The minimum time we should allow for Alice to transition from
+/// `current_state` to done.
+fn period_for_alice_to_complete(config: &Config, current_state: AliceState) -> Duration {
+    // Tail call recursion with an accumulator.
+    fn period_to_complete(config: &Config, state: AliceState, acc: Duration) -> Duration {
+        if state == AliceState::Done {
+            return acc;
+        }
+
+        let (_action, next_state) = state.next();
+        let transition_period = state.transition_period(config);
+
+        period_to_complete(config, next_state, acc + transition_period)
+    }
+
+    period_to_complete(config, current_state, Duration::zero())
+}
+
+/// The minimum time we should allow for Bob to transition from
+/// `current_state` to done.
+fn period_for_bob_to_complete(config: &Config, current_state: BobState) -> Duration {
+    // Tail call recursion with an accumulator.
+    fn period_to_complete(config: &Config, state: BobState, acc: Duration) -> Duration {
+        if state == BobState::Done {
+            return acc;
+        }
+
+        let (_action, next_state) = state.next();
+        let transition_period = state.transition_period(config);
+
+        period_to_complete(config, next_state, acc + transition_period)
+    }
+
+    period_to_complete(config, current_state, Duration::zero())
+}
+
+/// Happy path states for Alice.
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum AliceState {
+    // FIXME: Do we need a pre-created state in order to give Alice one more TTA period
+    // before `created`?
+    /// Initial state, swap has been created.
+    Created,
+    /// The fund alpha transaction has been sent to the network.
+    FundAlphaTransactionSent,
+    /// Implies fund alpha transaction has reached finality.
+    AlphaFunded,
```

Comment:
`Reach finality` is not a terminology we used before or commonly used. I would use "confirmed" instead.
```suggestion
    /// Implies fund alpha transaction has reached the target confirmation number
    FundAlphaTransactionConfirmed,
```

Reply:
@bonomat pointed out to me early in discussion that 'finality' is used. Here is an example search result: https://academy.binance.com/glossary/finality

Reply:
TIL.